### PR TITLE
feat: add support for computed property name

### LINKED ISSUE #698: wrong export of literal type
TS interface
```
const a = 'test-key';
export interface TagArray {
    [a]: string
}
```
is exported to
```
  "definitions": {
    "TagArray": {
      "additionalProperties": false,
      "properties": {
        "[a]": {
          "type": "string"
        }
      },
      "required": [
        "[a]"
      ],
      "type": "object"
    }
  }
```

`[a]` does not get resolved to `test-key` in the exported schema

### PR DESCRIPTION
**Pull request by alexts0**
fixes #698

### CODE REVIEW

File: test/valid-data/interface-computed-property-name/schema.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "$ref":        "#/definitions/MyObject",
+    "$schema":     "http://json-schema.org/draft-07/schema#",
+    "definitions": {
+        "MyObject": {
+            "type":                 "object",
```
Comment (domoritz) [inline test/valid-data/interface-computed-property-name/schema.json:6]: Please generate this schema so it's formatted like the other ones.

File: src/NodeParser/InterfaceAndClassNodeParser.ts [hunk 2]
```diff
@@ -164,4 +164,15 @@ export class InterfaceAndClassNodeParser implements SubNodeParser {
...
+
+    private getPropertyName(propertyName: PropertyName): string {
+        let result = propertyName.getText();
+        if (propertyName.kind === ts.SyntaxKind.ComputedPropertyName) {
+            const symbol = this.typeChecker.getSymbolAtLocation(propertyName);
+            if (symbol) {
+                result = symbol.getName();
+            }
+        }
+        return result;
```
Comment (domoritz) [inline src/NodeParser/InterfaceAndClassNodeParser.ts:176]:
```suggestion
    private getPropertyName(propertyName: PropertyName): string {
        if (propertyName.kind === ts.SyntaxKind.ComputedPropertyName) {
            const symbol = this.typeChecker.getSymbolAtLocation(propertyName);
            if (symbol) {
                return symbol.getName();
            }
        }
        return propertyName.getText()
```

**Review by domoritz — APPROVED** [review]
Thank you. Just added a small suggestion for refactoring.

**Comment by satvik-s** [comment]
Thank you!

**Pull request merged**